packages/atomic/src/components/commerce/facets/atomic-commerce-timeframe-facet/atomic-commerce-timeframe-facet.tsx

```diff
@@ -0,0 +1,317 @@
+import {parseDate} from '../../../../utils/date-utils';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {shouldDisplayInputForFacetRange} from '../../../common/facets/facet-common';
+import {FacetInfo} from '../../../common/facets/facet-common-store';
+import {FacetContainer} from '../../../common/facets/facet-container/facet-container';
+import {FacetGuard} from '../../../common/facets/facet-guard';
+import {FacetHeader} from '../../../common/facets/facet-header/facet-header';
+import {FacetPlaceholder} from '../../../common/facets/facet-placeholder/facet-placeholder';
+import {FacetValueLabelHighlight} from '../../../common/facets/facet-value-label-highlight/facet-value-label-highlight';
+import {FacetValueLink} from '../../../common/facets/facet-value-link/facet-value-link';
+import {FacetValuesGroup} from '../../../common/facets/facet-values-group/facet-values-group';
+import {initializePopover} from '../../../search/facets/atomic-popover/popover-type';
+import {CommerceBindings as Bindings} from '../../atomic-commerce-interface/atomic-commerce-interface';
+
+/**
+ * A facet is a list of values for a certain field occurring in the results.
+ * An `atomic-commerce-timeframe-facet` displays a facet of the results for the current query as date intervals.
+ *
+ * @internal
+ */
+@Component({
+  tag: 'atomic-commerce-timeframe-facet',
+  styleUrl: './atomic-commerce-timeframe-facet.pcss',
+  shadow: true,
+})
+export class AtomicCommerceTimeframeFacet
+  implements InitializableComponent<Bindings>
+{
+  @InitializeBindings() public bindings!: Bindings;
+  public facetForDateRange?: DateFacet;
+  public facetForDatePicker?: DateFacet;
+
+  @Element() private host!: HTMLElement;
+
+  /**
+   * The summary controller instance.
+   */
+  @Prop() summary!: SearchSummary | ListingSummary;
+  /**
+   * The date facet controller instance.
+   */
+  @Prop() public facet!: DateFacet;
+
+  @BindStateToController('facet')
+  @State()
+  public facetState?: DateFacetState;
+  @State() public error!: Error;
+
+  @State() private isCollapsed = false;
+  @State() private inputRange?: DateFilterRange;
+
+  private headerFocus?: FocusTargetController;
+
+  private get state() {
+    return this.facetState;
+  }
+
+  private get displayName() {
+    return this.state?.displayName || 'no-label';
+  }
+
+  private get focusTarget(): FocusTargetController {
+    if (!this.headerFocus) {
+      this.headerFocus = new FocusTargetController(this);
+    }
+    return this.headerFocus;
+  }
+
+  public initialize() {
+    this.registerFacetToStore();
+  }
+
+  @Listen('atomic/dateInputApply')
+  public applyDateInput({detail}: CustomEvent<DateFilterRange>) {
+    this.inputRange = {start: detail.start, end: detail.end};
+  }
+
+  private get valuesToRender() {
+    return (
+      this.facet?.state.values.filter(
```

Comment:
Out of curiosity, in general is there a reason why we're using `this.facet?.state.<...>` rather than `this.facetState?.<...>` when accessing the facet state within the component?